docs/src/modules/components/MarkdownDocs.js

```diff
@@ -103,6 +103,18 @@ export default function MarkdownDocs(props) {
           }
 
           const name = renderedMarkdownOrDemo.demo;
+
+          if (renderedMarkdownOrDemo.excludeByCurrentColorMode) {
+            const matchedColorMode = name.match(/([^/.]+(?=\.js|jsx|ts|tsx$))/)[0];
+
+            if (
+              ['light', 'dark'].includes(matchedColorMode) &&
+              matchedColorMode !== theme.palette.mode
+            ) {
+              return <span key={`${index}-excluded-by-current-color-mode`} />;
+            }
+          }
```

Comment:
I would like to only show `Demo.light.js` and exclude `Demo.dark.js` if the docs is currently in light mode (and vice versa)

Can anybody think of a less hacky way or is this more or less OK 😅 ? @mui/core

Reply:
I don't need to do this anymore